### PR TITLE
add stake distribution local state query

### PR DESCRIPTION
**Pull request by falcucci**
fixes https://github.com/txpipe/pallas/issues/316

### CODE REVIEW

File: pallas-network/Cargo.toml [hunk 1]
```diff
@@ -18,9 +18,15 @@ pallas-codec = { version = "=0.20.0", path = "../pallas-codec" }
 pallas-crypto = { version = "=0.20.0", path = "../pallas-crypto" }
 thiserror = "1.0.31"
 tokio = { version = "1", features = ["rt", "net", "io-util", "time", "sync", "macros"] }
+serde = { version = "1.0.136", optional = true, features = ["derive"] }
+serde_json = { version = "1.0.79", optional = true }
 tracing = "0.1.37"
```
Comment (scarmuega) [inline pallas-network/Cargo.toml:23]: why do we need this new dependencies?

File: pallas-network/src/miniprotocols/localstate/queries_v16/mod.rs [hunk 2]
```diff
@@ -80,6 +84,30 @@ pub struct SystemStart {
     pub picoseconds_of_day: u64,
 }
 
+#[derive(Debug, Encode, Decode, PartialEq)]
+pub struct StakeDistribution {
+    #[n(0)]
+    pub pools: KeyValuePairs<Bytes, Pool>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Encode, Decode, PartialEq, Clone)]
```
Comment (scarmuega) [inline pallas-network/src/miniprotocols/localstate/queries_v16/mod.rs:93]: why do we need Serialize / Deserialize on these structs?

Comment (falcucci) [reply pallas-network/src/miniprotocols/localstate/queries_v16/mod.rs:93]: nice catch, this is debug dirty stuff. removing

File: pallas-network/Cargo.toml (hunk 1 above)
Comment (falcucci) [reply pallas-network/Cargo.toml:23]: debug dirty stuff. removing

thanks!

**Review by scarmuega — APPROVED** [review]

**Pull request merged**